esi_leap/api/controllers/v1/node.py

```diff
@@ -64,14 +68,27 @@ def get_all(self):
         now = datetime.now()
         for node in nodes:
             offers = offer_obj.Offer.get_all({"resource_uuid": node.uuid,
-                                              "status": "available",
-                                              "start_time": now,
-                                              "end_time": now},
+                                              "status": "available"},
                                              context)
+            future_offers = []
+            for offer in offers:
+                if offer.start_time > now:
+                    future_offers.append(offer.uuid)
+                elif offer.end_time >= now:
+                    current_offer = offer
+            future_offers = ' '.join(future_offers)
+
+            f_leases = lease_obj.Lease.get_all({"resource_uuid": node.uuid,
+                                                "status": [statuses.CREATED]},
+                                               context)
+            f_lease_uuids = ' '.join([o.uuid for o in f_leases])
+
             n = Node(name=node.name, uuid=node.uuid,
-                     owner=keystone.get_project_name(node.owner, project_list))
-            if offers:
-                n.offer_uuid = offers[0].uuid
+                     owner=keystone.get_project_name(node.owner, project_list),
+                     future_offers=future_offers,
+                     future_leases=f_lease_uuids)
+            if 'current_offer' in locals():
```

Comment:
I think the more standard way to do this is to define ``current_offer = None`` up above, and then check ``if current_offer:`` here.

Reply:
Done